src/app/project/ontology/ontology.service.ts

```diff
@@ -79,4 +95,92 @@ export class OntologyService {
+                    // the property is not a subproperty of knora base ontology
+                    // get property iri from another ontology
+                    this._cache.get('currentProjectOntologies').subscribe(
+                        (ontologies: ReadOntology[]) => {
+                            const onto = ontologies.find(i => i.id === baseOntoIri);
+                            superPropIri = onto.properties[subProp].subPropertyOf[0];
+                        }
+                    );
+                }
+
+                if (superPropIri) {
+                    break;
+                }
+            }
+        }
+
+        return (superPropIri ? superPropIri : undefined);
+    }
+
+    /**
+     * get default property information for a certain ontology property
+     */
+    getDefaultPropType(property: ResourcePropertyDefinitionWithAllLanguages): Observable<DefaultProperty> {
+        let propType: DefaultProperty;
+
+        if (!property.guiElement) {
+            // we don't know what element to use, so it's unsupported property
+            return of (DefaultProperties.unsupported);
+        }
+
+        for (const group of this.defaultProperties) {
+            if (property.subPropertyOf.length) {
+                for (const subProp of property.subPropertyOf) {
+                    // if subProp is of type "link to" or "part of" we have to check the subproperty;
+                    // otherwise we get the necessary property info from the objectType
+                    if (subProp === Constants.HasLinkTo || subProp === Constants.IsPartOf) {
+                        propType = (group.elements.find(i =>
+                            i.guiEle === property.guiElement && i.subPropOf === subProp
+                        ));
+                    } else {
+
+                        // if the property is type of number or list, the gui element is not relevant
+                        // because the app supports only one gui element (at the moment): the spinbox resp. the list pulldown
+                        if (property.objectType === Constants.DecimalValue || property.objectType === Constants.ListValue) {
+                            propType = (group.elements.find(i =>
+                                i.objectType === property.objectType
+                            ));
+                        } else if (property.objectType === Constants.IntValue && subProp === Constants.SeqNum) {
+                            propType = (group.elements.find(i =>
+                                i.objectType === property.objectType && i.subPropOf === Constants.SeqNum
+                            ));
+                        } else {
+                            propType = (group.elements.find(i =>
+                                i.guiEle === property.guiElement && i.objectType === property.objectType
+                            ));
+                        }
+
+                    }
+                }
+                if (propType) {
```

Comment:
same as my comment above, maybe I'm missing something..

Reply:
As soon as the propType is defined (or found), we can exit the for-loop.